environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/ResourceEncryptionCreationHandler.java

```diff
@@ -0,0 +1,87 @@
+package com.sequenceiq.environment.environment.flow.creation.handler;
+
+import static com.sequenceiq.cloudbreak.common.mappable.CloudPlatform.AZURE;
+import static com.sequenceiq.environment.environment.flow.creation.event.EnvCreationHandlerSelectors.CREATE_ENVIRONMENT_RESOURCE_ENCRYPTION_EVENT;
+import static com.sequenceiq.environment.environment.flow.creation.event.EnvCreationStateSelectors.START_FREEIPA_CREATION_EVENT;
+
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.apache.commons.lang3.StringUtils;
+
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.environment.flow.creation.event.EnvCreationEvent;
+import com.sequenceiq.environment.environment.flow.creation.event.EnvCreationFailureEvent;
+import com.sequenceiq.environment.environment.service.EnvironmentService;
+import com.sequenceiq.flow.reactor.api.event.EventSender;
+import com.sequenceiq.flow.reactor.api.handler.EventSenderAwareHandler;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class ResourceEncryptionCreationHandler extends EventSenderAwareHandler<EnvironmentDto> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ResourceEncryptionCreationHandler.class);
+
+    private final EnvironmentService environmentService;
+
+    private final EntitlementService entitlementService;
+
+    private final EventBus eventBus;
+
+    protected ResourceEncryptionCreationHandler(EventSender eventSender, EnvironmentService environmentService, EventBus eventBus,
+            EntitlementService entitlementService) {
+        super(eventSender);
+        this.environmentService = environmentService;
+        this.eventBus = eventBus;
+        this.entitlementService = entitlementService;
+    }
+
+    @Override
+    public String selector() {
+        return CREATE_ENVIRONMENT_RESOURCE_ENCRYPTION_EVENT.selector();
+    }
+
+    @Override
+    public void accept(Event<EnvironmentDto> environmentDtoEvent) {
+        LOGGER.debug("Accepting ResourceEncryptionInitialization event");
+        EnvironmentDto environmentDto = environmentDtoEvent.getData();
+        try {
+            environmentService.findEnvironmentById(environmentDto.getId()).ifPresent(environment -> {
+                if (AZURE.name().equalsIgnoreCase(environmentDto.getCloudPlatform())) {
+                    String encryptionKeyUrl = Optional.ofNullable(environmentDto.getParameters())
+                            .map(paramsDto -> paramsDto.getAzureParametersDto())
```

Comment:
is this nullpointer safe?

Reply:
yes, will eventually be `null` https://github.com/hortonworks/cloudbreak/pull/10308/files/93366b4ccb10dff4b233e32603981542226b2fc7#diff-8769f5fa729224732dc57338cd75e9295c138c3e8ca7811b1751469d6ee68651R58